PracticeGit/intlist.c

```diff
@@ -0,0 +1,170 @@
+//------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+//-------------------------------------------------------------------------------------------------------------------------------------------------------------------
+//   intlist.c
+//   Program on Linked List functions.
+//------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdlib.h>
+#include "intlist.h"
+
+// Error message for memory allocation failure
+const char* memoryAllocationError = "Memory allocation failed.\n";
+// Initialize the head pointer of the linked list to NULL.
+struct node* head = NULL;
+
+// Function to create a empty linked list
+struct node* CreateList(int initialData) {
+   struct node* head = (struct node*)malloc(sizeof(struct node));
+   if (head == NULL) {
+      printf("%s", memoryAllocationError);
+      exit(1);

```

Comment:
Why exit the program itself? This is not proper behaviour. An error should be returned so that the calling program can make a decision.